vignettes/dev-guide/dg_printing.Rmd

```diff
@@ -0,0 +1,288 @@
+    )
+    # If column counts are displayed, add column count format
+    if (disp_ccounts(obj)) {
+      hdr_fmt_blank[nrow(hdr_fmt_blank), ] <- c("", rep(colcount_format(obj), ncol(obj)))
+    }
+
+    formats <- rbind(hdr_fmt_blank, formats_strings)
+
+    spans <- rbind(header_content$span, body_spans)
+    row.names(spans) <- NULL
+
+    aligns <- rbind(
+      matrix(rep("center", length(header_content$body)),
+        nrow = nrow(header_content$body)
+      ),
+      body_aligns
+    )
+
+    aligns[, 1] <- "left" # row names and topleft (still needed for topleft)
+
+    # Main indentation of the table rownames #-> Main indentation facility
+    if (indent_rownames) {
+      body[, 1] <- indent_string(body[, 1], c(rep(0, nr_header), sr$indent),
+        incr = indent_size
+      )
+      formats[, 1] <- indent_string(formats[, 1], c(rep(0, nr_header), sr$indent),
+        incr = indent_size
+      )
+    }
+    
+    #-> referential strings are added to the table. get_ref_matrix is the core of this process
+    #   along with format_fnote_ref that in this case is used to format the reference string and their
+    #   indices. Note that the footnotes for the header is taken from the output of .tbl_header_mat
+    # Handling of references in header and body
+    col_ref_strs <- matrix(vapply(header_content$footnotes, function(x) {
+      if (length(x) == 0) {
+        ""
+      } else {
+        paste(vapply(x, format_fnote_ref, ""), collapse = " ")
+      }
+    }, ""), ncol = ncol(body))
+    body_ref_strs <- get_ref_matrix(obj)
+    body <- matrix(
+      paste0(
+        body,
+        rbind(
+          col_ref_strs, #-> col_ref_strs are added to the body as a separate section
+          body_ref_strs
+        )
+      ),
+      nrow = nrow(body),
+      ncol = ncol(body)
+    )
+    
+    # Solve \n in titles
+    if (any(grepl("\n", all_titles(obj)))) {
+      if (any(grepl("\n", main_title(obj)))) {
+        tmp_title_vec <- .quick_handle_nl(main_title(obj))
+        main_title(obj) <- tmp_title_vec[1]
+        subtitles(obj) <- c(tmp_title_vec[-1], .quick_handle_nl(subtitles(obj)))
+      } else {
+        subtitles(obj) <- .quick_handle_nl(subtitles(obj))
+      }
+    }
+
+    # Solve \n in footers
+    main_footer(obj) <- .quick_handle_nl(main_footer(obj))
+    prov_footer(obj) <- .quick_handle_nl(prov_footer(obj))
+
+    
+    #-> this is still under development as indicated by xxx. The idea is to allow \n also in peculiar
+    #   cases, such as page titles and referential footnotes. The latter are resolved in toString (pagination
+    #   will not count them as more than one line each), while for the former we do not have any coverage yet.
+    # xxx \n in page titles are not working atm (I think)
+    # ref_fnotes <- strsplit(get_formatted_fnotes(obj), "\n", fixed = TRUE)
+    ref_fnotes <- get_formatted_fnotes(obj) # pagination will not count extra lines coming from here
+    pag_titles <- page_titles(obj)
```

Comment:
I think this needs to be a stop-error for now. I do not know precisely how to correct pagination for this (solving refnotes here breaks toString; we need to change the structure of ref_notes otherwise, not super easy). For sure I will add coverage for these two cases this week (the first one is solved but the line count for pagination is wrong)